packages/core/src/index.ts

```diff
@@ -4,7 +4,7 @@ export type Context<T> = {};
 
 // These compile away
 export const useState = <T>(obj: T) => obj;
-export const useRef = () => null as any;
+export const useRef = <T>(obj?: null | undefined | T) => obj as unknown as T;
```

Comment:
@samijaber I need some typing help here. I want useRef to be valid if no arguments or type params are added

`const something = useRef()`

Reply:
Ugh, wrote a comment but it got erased somehow

Why do we want to make no-arguments a valid format? A reference needs to have some sort of default initial value, and it is far better for the end-user to provide that explicitly, rather than we do something implicit and then fiddle with types etc.

That would follow React's API as well: whatever type is provided to `useRef` is the type of the reference, and the user has to match that:

```suggestion
export const useRef = <T>(obj:T) => obj;
```

Reply:
It will also vastly simplify our parser and generators code, which in this PR involve a bunch of code to figure out if this is a dom ref or not, to then figure out that an initial `null` needs to be set. 

I think we should just do what React does and keep things simple.

Reply:
React allows for undefined its just for DOM refs it should be set to null

Reply:
I missed the notification for this last comment.

Do you still want to allow no args if the type args are specified?